File: kata-solutions/bugs-zero-kata/bugs-zero-python/src/game_runner.py
from random import seed, randrange
from game import Game, Player


class GameRunner:
    def __init__(self):
        self.not_a_winner = False

    @staticmethod
    def main():
        game = GameRunner()
        seed(30)
        game.play_game(randrange(5) + 1)

    def play_game(self, rand:int) -> None:
        game = Game(Player('Chet'), Player('Pat'), [Player('Sue')])

        while True:
            game.roll(rand)

            if randrange(9) == 7:
                self.not_a_winner = game.wrong_answer()
            else:
                self.not_a_winner = game.was_correctly_answered()

            if not self.not_a_winner:
                break


if __name__ == '__main__':
    GameRunner.main()
    

File: kata-solutions/bugs-zero-kata/bugs-zero-typescript/src/GameRunner.ts
'use strict';

import { Game, Player } from "./Game"
import seedrandom from "seedrandom";

export class GameRunner {
    private notAWinner: boolean;
    private randomNumberGenerator = seedrandom('1234');

    public constructor() {
        this.notAWinner = false;
    }

    public doRoll(): number {
        return Math.floor(this.randomNumberGenerator() * 6) + 1;
    }

    public playGame(rand: number) {
        var game = new Game(new Player('Chet'), new Player('Pat'), [new Player('Sue')]);

        while (true) {
            game.roll(rand);

            if (Math.floor(this.randomNumberGenerator() * 10) == 7)
                this.notAWinner = game.wrongAnswer();
            else
                this.notAWinner = game.wasCorrectlyAnswered();

            if (!this.notAWinner)
                break;
        }
    }
}
